### DISCUSSION TITLE
trivy db update incremental or full?

### DISCUSSION THREAD

**Discussion by ktzsolt** [aquasecurity/trivy#7506] [Q&A] [answered]
### Question

Hi!

I would like to know whether these commands do full db download and update or just incremental download and update?
```
trivy image --download-db-only
trivy image --download-java-db-only
```

I am planning to do this in CI and I don't want to download daily the full db if not necessary so I would utilize docker build cache and do the full db update only weekly by invalidating the layers.

Thanks!

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Operating System

ubuntu

### Version

```bash
0.55.1
```

**Comment by knqyf263** [comment, marked as answer]
It's full. Note that the compressed DB size is ~50MB.

**Reply by ktzsolt** [reply]
Thanks! 

I need the java db too with `trivy image --download-java-db-only ` too, that is 700 Mb. I will update my trivy image every day or something like that and use it in CI for scans and before a release build I will make a force update.
